### ISSUE TITLE
[CI] Browser tests are failing in macos-latest due to Puppeteer

### ISSUE THREAD

**Issue by smoya** [asyncapi/parser-js#1001]
### Description

Current browser tests are broken in MacOS due to a timeout. See the following run as example: https://github.com/asyncapi/parser-js/actions/runs/9348904713/job/25775243162?pr=999#step:10:443

![Google Chrome_PikQRSCy](https://github.com/asyncapi/parser-js/assets/1083296/e60647f0-c896-4fda-a910-8060c55ddec0)

I've tried increasing timeouts, but it always ends with such a timeout.
After a deep investigation, I discovered the issue happens when trying to start a new browser instance through **Puppeteer**. 

### But why this is happening suddenly? 

`macos-latest` GitHub-hosted runner tag was migrated (through a [12 weeks of rolling update](https://github.blog/changelog/2024-04-01-macos-14-sonoma-is-generally-available-and-the-latest-macos-runner-image/)) from pointing to `macos-12` to `macos-14`. Not that exactly, but in practical effects, that's what happened.
The issue comes with this new `macos-14` version: **`macos-14` is now on Apple Silicon (arm64) and not Intel anymore.**. 

### Steps I follow but didn't fix the issue

1. I decided to upgrade **Puppeteer** to latest version because I saw there was people complaining about an incompatibility with M2/M3 Apple chipsets. Suddenly tests on Linux were also failing due to the same timeout.

2. TODO

### Possible solutions

1. To downgrade to `macos-13`, at least until we find the issue and a remediation. That would require modifying the global workflow file at https://github.com/asyncapi/.github/blob/master/.github/workflows/if-nodejs-pr-testing.yml

cc @derberg

**Comment by smoya** [comment]
Ok, I found out that Puppeteer GH project downgraded from `macos-latest` to `macos-13` 😅 . See https://github.com/puppeteer/puppeteer/pull/12326

They didn't provide any description on the PR or in the issue, so the reasons are not there but I can imagine they faced this issue so they decided to not provide support to macos 14? They have an opened issue to move back to latest version. I wrote a comment there: https://github.com/puppeteer/puppeteer/issues/12327#issuecomment-2147617895

I guess we should downgrade to `macos-13` in the meantime. @derberg
